### PR TITLE
chore(semantic-release) - add semantic release

### PR DESCRIPTION
**Pull request by rdotg**
This is the base work to add semantic release as a GitHub action.

It will probably take a few tries to get this right.

Once we merge this to main, it will kick off the action and create a new environment where we can add an npm token and a gh token: https://github.com/silinternational/ui-components/settings/environments

Then to test we can push a change to with the proper commit prefix to trigger a deploy.

Here are the appropriate prefixes: 

Commit message | Release type
-- | --
fix(pencil): stop graphite breaking when too much pressure applied | Patch Release
feat(pencil): add 'graphiteWidth' option | Minor Feature Release
perf(pencil): remove graphiteWidth option  BREAKING CHANGE: The graphiteWidth option has been removed. The default graphite width of 10mm is always used for performance reasons. | Major Breaking Release

If you don't use these, the GitHub actions will have no effect. IE: It won't deploy to NPM, and it won't update GitHub with the updated version in package.json or add a release to the release tab.

### CODE REVIEW

**Review by hobbitronics — COMMENTED** [review]
The lock file here is much smaller and version 1, presumably you built it with an older version of npm. I see version 12 is used in the action. I was able to build it with a later version 7.20.0 and it results in a larger lock file similar to what we had before. Should we go with older or newer version of npm for the lockfile build?

**Review by Schparky — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        uses: actions/checkout@v1
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Install dependencies
+        run: npm ci
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (hobbitronics) [inline .github/workflows/release.yml:21]: Failed to add secret. Secret names must not start with GITHUB_

Comment (hobbitronics) [reply .github/workflows/release.yml:21]:
```suggestion
          GH_TOKEN: ${{ secrets.GH_TOKEN }}
```

**Review by hobbitronics — APPROVED** [review]
Looks good except the comment I made about the github token name not allowed.

**Pull request merged**